extensions/optaplanner/runtime/src/main/java/io/quarkus/optaplanner/OptaPlannerRecorder.java

```diff
@@ -11,6 +11,10 @@
 
     public BeanContainerListener initialize(SolverConfig solverConfig, SolverManagerConfig solverManagerConfig) {
         return container -> {
+            if (solverConfig.getSolutionClass() != null) {
```

Comment:
Is this test correct?

Reply:
If the solution Class is zero, it will fail fast a bit later anyway, with a decent error message.
It's to avoid eating that decent error message with an NPE here.

But I guess the build classloader should never excape to the runtime? I can also set the classloader to null in that corner case? Wdyt?

Reply:
Yeah, that sounds fine to me.